### PR TITLE
Run intl-data tests on resources change

### PR DESCRIPTION
**Pull request by jakzal**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? |no
| Tickets       | -
| License       | MIT
| Doc PR        | -

`intl-data` tests are run whenever we update ICU data as they're rather time-consuming.

This proposal automates intl-data test runs with Github Actions. Tests will only be run on pull requests or pushes that modifyy ICU data. Full run on Github Actions takes 6-7mins. Example run: https://github.com/jakzal/symfony/actions/runs/406526765

image

### CODE REVIEW

File: .github/workflows/intl-data-tests.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+                with:
+                    fetch-depth: 2
+
+            -   name: Check if intl data files were modified
+                id: intl-data-check
+                run: |
+                    (git diff --name-only HEAD^ HEAD | grep src/Symfony/Component/Intl/Resources/data) &&
```
Comment (OskarStark) [inline .github/workflows/intl-data-tests.yml]: Couldn't this be done on top 🔝 using:

https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestpaths

?

**Comment by OskarStark** [comment]
Hi @jakzal it looks like the commit is not associated with your github account:
image

File: .github/workflows/intl-data-tests.yml (hunk 1 above)
Comment (jakzal) [reply .github/workflows/intl-data-tests.yml]: Nope. That would make that tests are always executed as these paths always exist.

Comment (jakzal) [reply .github/workflows/intl-data-tests.yml]: However, reading the following paragraph, it should work. I've tested it and it didn't... let me try again

image

Comment (jakzal) [reply .github/workflows/intl-data-tests.yml]: Yeah, that worked this time... not sure what I did wrong the first round. Thanks for making me try it again!

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jakzal.

File: .github/workflows/intl-data-tests.yml (hunk 1 above)
Comment (OskarStark) [reply .github/workflows/intl-data-tests.yml]: Great!
